distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso

```diff
@@ -178,10 +202,20 @@ limit_data limit data = case limit of
         if limit <= extreme.length then extreme.take (..First limit) else
             extreme + data.take (..Sample (limit - extreme.length))
 
+
+## PRIVATE
+add_fields : Builder -> Integer -> Integer -> Vector | Nothing
+add_fields current number_of_numeric idx=0 = if idx == number_of_numeric then current.to_vector else
+    current.append (Point_Data.Y_multi idx)
+    @Tail_Call add_fields current number_of_numeric idx+1
+
 ## PRIVATE
 json_from_table : Table -> Vector Integer | Nothing -> Integer | Nothing -> Text
 json_from_table table bounds limit =
-    data = table.point_data |> bound_data bounds |> limit_data limit
+    number_of_numeric_cols = table.multi_series_info-2
+    feilds_for_multiseries = Point_Data.all_fields + add_fields (Builder.new number_of_numeric_cols) number_of_numeric_cols
```

Comment:
looks like a typo :)
```suggestion
    fields_for_multiseries = Point_Data.all_fields + add_fields (Builder.new number_of_numeric_cols) number_of_numeric_cols
```

Reply:
In general the raw `Vector.Builder.new` should be used only rarely. Normally we'd prefer to use `Vector.build` idiom.

But here I think we can use just a simple range-iterator:
```suggestion
    feilds_for_multiseries = Point_Data.all_fields + ((0.up_to number_of_numeric_cols).map idx-> Point_Data.Y_multi idx)
```

Unless I misunderstood your intent here - we want to add `Point_Data.Y_multi idx` for each index from `0` to `number_of_numeric_cols`, right? Then using `up_to` with `map` is the most idiomatic way to do it.

Or some may prefer:
```suggestion
    feilds_for_multiseries = Point_Data.all_fields + (Vector.new number_of_numeric_cols idx-> Point_Data.Y_multi idx)
```

but IMHO `up_to` is a bit clearer